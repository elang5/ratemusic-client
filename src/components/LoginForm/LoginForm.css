.login-form {
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
  text-align: center;
  width: 65%;
  height: 75%;
  border-radius: 5.5px;
  box-shadow: 0 0 3px 2px rgba(0, 0, 0, 0.15);
  background-color: #ffffff;
}

.sign-in {
  color: #1f1f21;
  font-size: 24px;
}

.label {
  font-size: 18px;
  color: #1f1f21;
}

#user_name {
  margin-bottom: .5em;
}

#user_name, #password {
  margin-top: 1em;
  width: 41.5%;
  padding: 15px 20px;
  border-radius: 2.2px;
  border: solid 0.6px rgba(208, 208, 208, 0.5);
  background-color: #e5e9ed;
  font-size: 20px;
}


.submit-btn {
  display: block;
  width: 41.5%;
  padding: 15px 20px;
  background-color: rgb(252, 60, 57);
  margin: 2rem auto;
  border-radius: 2.2px;
  font-size: 20px;
  color: white;
  font-weight: 200;
}
