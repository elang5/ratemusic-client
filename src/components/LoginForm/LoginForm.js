import React, { Component } from 'react'
import TokenService from '../../services/token-service'
import AuthApiService from '../../services/auth-api-service'
import './LoginForm.css'

export class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = {
    error: null
  }

  handleSubmitJwtAuth = e => {
    e.preventDefault()
    const { user_name, password } = e.target

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
      .then(res => {
        user_name.value = ''
        password.value = ''
        TokenService.saveAuthToken(res.authToken)
        this.props.onLoginSuccess()
      })
      .catch(err => {
        this.setState({ error: err.error })
      })
  }

  handleDemoLogin = () => {
    AuthApiService.postLogin({
      user_name: 'testing',
      password: '!Testing1'
    })
      .then(res => {
        TokenService.saveAuthToken(res.authToken)
        this.props.onLoginSuccess()
      })
      .catch(err => {
        this.setState({ error: err.error })
      })
  }
  render() {
    const { error } = this.state
    return (
        <form 
          className="login-form"
          onSubmit={this.handleSubmitJwtAuth}
        >
          <p className="sign-in">Sign In To Rate Music</p>
          <div className="error-alert">
            {error && <p className="error">{error}</p>}
          </div>
          <div className="user_name_input">
            <label 
              className="label" 
              htmlFor="user_name">
              Username
            </label><br />
            <input 
              required 
              placeholder="testing"
              type="text" 
              name="user_name" 
              id="user_name"/>
          </div>
          <div className="password_input">
            <label 
              className="label" 
              htmlFor="password">
              Password
            </label><br />
            <input 
              required 
              placeholder="!Testing1"
              type="password" 
              name="password" 
              id="password"/>
          </div>
          <div className="btns-container">
            <button 
              type="submit" 
              className="submit-btn">
              Sign In
            </button>
            <button
              type="button"
              className="submit-btn demo"
              onClick={this.handleDemoLogin}>
              Demo Login
            </button>
          </div>
        </form>
    )
  }
}

export default LoginForm
