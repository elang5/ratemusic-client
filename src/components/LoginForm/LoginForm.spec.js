import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginForm from './LoginForm'

describe(`LoginForm Component`, () => {

  it(`renders the LoginForm component`, () => {
    const wrapper = shallow(<LoginForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})