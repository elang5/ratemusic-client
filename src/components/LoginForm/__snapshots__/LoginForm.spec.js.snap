// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component renders the LoginForm component 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <p
    className="sign-in"
  >
    Sign In To Rate Music
  </p>
  <div
    className="error-alert"
  />
  <div
    className="user_name_input"
  >
    <label
      className="label"
      htmlFor="user_name"
    >
      Username
    </label>
    <br />
    <input
      id="user_name"
      name="user_name"
      placeholder="testing"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password_input"
  >
    <label
      className="label"
      htmlFor="password"
    >
      Password
    </label>
    <br />
    <input
      id="password"
      name="password"
      placeholder="!Testing1"
      required={true}
      type="password"
    />
  </div>
  <div
    className="container"
  >
    <button
      className="submit-btn"
      type="submit"
    >
      Sign In
    </button>
  </div>
</form>
`;
